packages/components/filters/src/filter-menu/header/header.tsx

```diff
@@ -1,5 +1,73 @@
-function Header() {
-  return <div>header</div>;
-}
+import { useState, ReactNode } from 'react';
+import IconButton from '@commercetools-uikit/icon-button';
+import Spacings from '@commercetools-uikit/spacings';
+import { SortingIcon } from '@commercetools-uikit/icons';
+import { designTokens } from '@commercetools-uikit/design-system';
+import { css } from '@emotion/react';
+
+type TOption = {
+  value: string;
+  label: string;
+};
+
+type THeaderProps = {
+  headerLabel: ReactNode;
+  operandOptions?: Array<TOption>;
+  renderOperandsInput?: boolean;
+  /**
+   * the function to render the operator input
+   */
+  renderOperatorsInput?: () => ReactNode;
+  /**
+   * the function to sort the data
+   */
+  onSort?: () => void;
+  /**
+   * the width of the menu header
+   */
+  menuHeaderWidth?: string;
+};
+
+const headerContainerStyles = css`
+  display: flex;
+  align-items: center;
+  padding-bottom: ${designTokens.spacing10};
+  border-bottom: 1px solid ${designTokens.colorNeutral90};
+`;
+
+const getSelectInputStyles = (props: THeaderProps) => css`
+  width: ${props.menuHeaderWidth};
+  margin: ${designTokens.spacing20};
+`;
+
+const Header = (props: THeaderProps) => {
+  const [isActive, setIsActive] = useState(false);
+
+  return (
+    <Spacings.Inline>
```

Comment:
`Spacings.Inline` has no effect, it can be safely ditched.  

![image](https://github.com/user-attachments/assets/574bfac3-b2fe-4464-9855-6b8ce20ebffe)

I feel like the `headerContainer` can consume the available space  and make sure the sort-button is always on the right. But the label- and input-div probably need some tinkering.

@ByronDWall can you give a bit of direction?

Can labels wrap, or do they have a fixed width/maxWidth? What happens if this width is reached? wrap lines, ellipsis? Does the select need its size manually set, or can it just occupy the leftover space?

Reply:
updated here: https://github.com/commercetools/ui-kit/pull/2951/commits/87f00d071d260cc8ba05dc38380993e55887aeca

Lets discuss this and I will follow up

Thank you

Reply:
the header should always take up 100% of the width of the container, and as long as there is an 8px gap between the label and the input, and a 24px gap between the input and the sort button, the operators input should be able to take up the remaining space. ([as specified in the figma](https://www.figma.com/design/b2S3GGpScVDuf15VAGEqeL/Filter-pattern?node-id=2040-8188&m=dev))

As far as min/max widths for the label, and whether it should truncate or wrap, I'd like to get clarification from @FilPob on that.  Same for min/max width for the operators input

Reply:
I put some more examples [here](https://www.figma.com/design/b2S3GGpScVDuf15VAGEqeL/Filter-pattern?m=auto&node-id=5581-34207&t=KR8QhepMF2i7dnoh-1). Hope that makes it clearer

- yes the header should always take 100% of the parent
- the sort icon should be always on the right edge of the parent
- the operator selector' width should be set to wrap it's content... it should not take the available space
- The filter name label should break into multiple lines if it's too long

Reply:
@FilPob  Thank you for clarifying, just want to clarify before I push the change. It should basically look like these right? 
 




<img width="372" alt="image" src="https://github.com/user-attachments/assets/5950a61b-4fea-4b09-af62-547970dcc3b2">

<img width="315" alt="image" src="https://github.com/user-attachments/assets/c7229bb3-cd98-4097-86fe-9a1f91186335">

<img width="336" alt="image" src="https://github.com/user-attachments/assets/bfe1aaa9-0ec4-4f6d-9425-41dd7835e68d">

Reply:
updated here: https://github.com/commercetools/ui-kit/pull/2951/commits/f0a8d312959a49036887fad0bb611df89e3fac1e

Reply:
yes that looks good :)